bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```
@@ -0,0 +1,684 @@
+     *
+     * @param date The date string read from the device
+     * @return A DateTimeType object representing the date or time specified
+     */
+    private DateTimeType toDateTime(String date) {
+        String[] dateTimeParts = null;
+        String dateTime = date;
+        try { // date
+            dateTimeParts = date.split("\\."); // try to split date components
+            if (dateTimeParts.length == 1) { // time
+                return DateTimeType.valueOf(date);
+            } else { // date - we'll try the device's date format
+                switch (this.dateFormat) {
+                    case 0: // dd.mm.yyyy
+                        dateTime = dateTimeParts[2] + "-" + dateTimeParts[1] + "-" + dateTimeParts[0];
+                        break;
+                    case 1: // mm.dd.yyyy
+                        dateTime = dateTimeParts[2] + "-" + dateTimeParts[0] + "-" + dateTimeParts[1];
+                        break;
+                    case 2: // yyyy.mm.dd
+                        dateTime = dateTimeParts[0] + "-" + dateTimeParts[1] + "-" + dateTimeParts[2];
+                        break;
+                    default:
+                        dateTime = dateTimeParts[2] + "-" + dateTimeParts[1] + "-" + dateTimeParts[0];
+                        break;
+                }
+                return DateTimeType.valueOf(dateTime);
+            }
+        } catch (Exception e) {
+            // falling back to default date format (apparently using the configured format has failed)
+            if (dateTimeParts != null) {
+                dateTime = dateTimeParts[2] + "-" + dateTimeParts[1] + "-" + dateTimeParts[0];
+                return DateTimeType.valueOf(dateTime);
+            } else {
+                logger.warn("{} couldn't parse the provided date string: {}",
+                        HeliosEasyControlsHandler.class.getSimpleName(), date);
+                return new DateTimeType(); // return at least a valid DateTimeType object

```

You could throw an exception here, instead of returning a dummy value.